src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -83,73 +118,748 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    this->actionFunc = EnFz_Despawn;
+}
+
+void EnFz_Despawn(EnFz* this, GlobalContext* globalCtx) {
+    if (this->timer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnFz_SetupMelt(EnFz* this) {
+    this->state = 3;
+    this->isFreezing = false;
+    this->isDespawning = true;
+    this->actor.flags &= ~1;
+    this->actionFunc = EnFz_Melt;
+    this->actor.speedXZ = 0.0f;
+    this->speedXZ = 0.0f;
+}
+
+void EnFz_Melt(EnFz* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.scale.y, 0.0006f, 0.0002f);
+
+    if (this->actor.scale.y < 0.006f) {
+        this->actor.scale.x += 0.0004f;
+        this->actor.scale.z += 0.0004f;
+    }
+
+    if (this->actor.scale.y < 0.004f) {
+        this->envAlpha -= 8;
+        if (this->envAlpha > 255) {
+            this->envAlpha = 0;
+        }
+    }
+
+    if (this->envAlpha == 0) {
+        EnFz_SetupDespawn(this, globalCtx);
+    }
+}
+
+void EnFz_SetupBlowSmokeStationary(EnFz* this) {
+    this->state = 1;
+    this->timer = 40;
+    this->updateBgInfo = true;
+    this->isFreezing = true;
+    this->actor.flags |= 1;
+    this->actionFunc = EnFz_BlowSmokeStationary;
+    this->actor.gravity = -1.0f;
+}
+
+void EnFz_BlowSmokeStationary(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 isTimerMod8;
+    s16 primAlpha;
+
+    if (this->counter & 0xC0) {
+        EnFz_SetYawTowardsPlayer(this);
+        EnFz_UpdateTargetPos(this, globalCtx);
+    } else {
+        isTimerMod8 = false;
+        primAlpha = 150;
+        func_8002F974(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
+
+        if ((this->counter & 0x3F) >= 48) {
+            primAlpha = 630 - ((this->counter & 0x3F) * 10);
+        }
+
+        accel.x = accel.z = 0.0f;
+        accel.y = 0.6f;
+
+        pos.x = this->actor.world.pos.x;
+        pos.y = this->actor.world.pos.y + 20.0f;
+        pos.z = this->actor.world.pos.z;
+
+        Matrix_RotateY((this->actor.shape.rot.y / (f32)0x8000) * M_PI, MTXMODE_NEW);
+
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f;
+
+        Matrix_MultVec3f(&vec1, &velocity);
+
+        if ((this->counter % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SpawnIceSmokeFreeze(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+
+        EnFz_SpawnIceSmokeFreeze(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
+    s32 pad;
+    static EnFzSpawnIceSmokeFunc iceSmokeSpawnFuncs[] = {
+        EnFz_SpawnIceSmokeHiddenState,
+        EnFz_SpawnIceSmokeGrowingState,
+        EnFz_SpawnIceSmokeActiveState,
+        EnFz_SpawnIceSmokeActiveState,
+    };
```

Comment:
I suggest moving this array above the function if it still matches. Otherwise it should probably be above the local variables